src/algorithms/algorithm.py

```diff
@@ -60,13 +58,11 @@ def find_path(self) -> None:
         Method for finding a path from agent to goal
         Movement should be done using the map APIs
         """
-        self._services.lock.acquire()
         if self.testing is not None:
             self.testing.algorithm_start()
         self._find_path_internal()
         if self.testing is not None:
             self.testing.algorithm_done()
-        self._services.lock.release()
```

Comment:
are we not using the lock any more?

Reply:
No more coarse grained locking, definitely just want to be using condition vars (which under the hood do use a lock). Main reason is because we'd then need to expose the lock to the algorithm to arbitrarily release and acquire the lock. This new way means we don't need to modify any algorithms whilst guaranteeing a smooth experience at all time.